File: Python/783. Minimum Distance Between BST Nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def minDiffInBST(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        a = []
        def dfs(node):
            if node is None:
                return None
            dfs(node.left)
            a.append(node.val)
            dfs(node.right)
        dfs(root)
        ans = -1
        for i in range(1, len(a)):
            if ans != -1:
                ans = min(ans, a[i]-a[i-1])
            else:
                ans = a[i] - a[i-1]
        return ans

File: C++/783. Minimum Distance Between BST Nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> a;
    void dfs(TreeNode* node)
    {
        if(node==nullptr)
            return ;
        dfs(node->left);
        a.push_back(node->val);
        dfs(node->right);
    }
    int minDiffInBST(TreeNode* root) {
        a.clear();    
        dfs(root);
        int ans = -1, len = a.size();
        for(int i=1;i<len;i++)
        {
            if(ans == -1)
                ans = a[i]-a[i-1];
            else
                ans = min(ans,a[i]-a[i-1]);
        }
        return ans;
    }
};